ApplicationLibCode/Commands/RicCreateSurfaceIntersectionBandFeature.cpp

```diff
@@ -0,0 +1,142 @@
+
+#include "Riu3DMainWindowTools.h"
+
+#include "cafSelectionManager.h"
+
+#include <QAction>
+
+CAF_CMD_SOURCE_INIT( RicCreateSurfaceIntersectionBandFeature, "RicCreateSurfaceIntersectionBandFeature" );
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimSurfaceCollection* RicCreateSurfaceIntersectionBandFeature::surfaceCollection()
+{
+    RimProject* proj = RimProject::current();
+    return proj->activeOilField()->surfaceCollection();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+bool RicCreateSurfaceIntersectionBandFeature::isCommandEnabled()
+{
+    auto* surfColl = RicCreateSurfaceIntersectionBandFeature::surfaceCollection();
+    auto  surfaces = surfColl->ensembleSurfaces();
+
+    return !surfaces.empty();
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RicCreateSurfaceIntersectionBandFeature::onActionTriggered( bool isChecked )
+{
+    auto* intersection = caf::SelectionManager::instance()->selectedItemAncestorOfType<RimExtrudedCurveIntersection>();
+    if ( intersection )
+    {
+        RimEnsembleSurface* firstEnsembleSurface = nullptr;
+        {
+            auto surfColl = RicCreateSurfaceIntersectionBandFeature::surfaceCollection();
+            auto surfaces = surfColl->ensembleSurfaces();
+            if ( !surfaces.empty() ) firstEnsembleSurface = surfaces.front();
+        }
+
+        RimSurfaceIntersectionBand* objectToSelect = nullptr;
+
+        if ( firstEnsembleSurface )
+        {
+            const double defaultOpacity = 0.6;
+
+            auto colors = RiaColorTables::structuralUncertaintyColors();
+
+            // Create min/max band
+            {
+                auto surf1 =
+                    firstEnsembleSurface->findStatisticsSurface( RigSurfaceStatisticsCalculator::StatisticsType::MIN );
+                auto surf2 =
+                    firstEnsembleSurface->findStatisticsSurface( RigSurfaceStatisticsCalculator::StatisticsType::MAX );
+
+                if ( surf2 && surf1 )
+                {
+                    auto band = intersection->addIntersectionBand();
+                    band->setSurfaces( surf1, surf2 );
+
+                    auto color = colors.cycledColor3f( 0 );
+                    band->setBandColor( color );
+                    band->setBandOpacity( defaultOpacity );
+                    band->setPolygonOffsetUnit( 80 );
+
+                    band->lineAppearance()->setColor( color );
+
+                    objectToSelect = band;
+                }
+            }
```

Comment:
Extract method?

Reply:
I suggest we leave this for now, there are too many parameters that are different